HousingSearchListener/V1/UseCase/PersonBalanceUpdatedUseCase.cs

```diff
@@ -0,0 +1,80 @@
+﻿using HousingSearchListener.V1.Boundary;
+using HousingSearchListener.V1.Domain.Account;
+using HousingSearchListener.V1.Domain.Person;
+using HousingSearchListener.V1.Domain.Tenure;
+using HousingSearchListener.V1.Factories;
+using HousingSearchListener.V1.Gateway;
+using HousingSearchListener.V1.Infrastructure.Exceptions;
+using HousingSearchListener.V1.UseCase.Interfaces;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace HousingSearchListener.V1.UseCase
+{
+    public class PersonBalanceUpdatedUseCase : IPersonBalanceUpdatedUseCase
+    {
+        private readonly IEsGateway _esGateway;
+        private readonly ITenureApiGateway _tenureApiGateway;
+        private readonly IPersonApiGateway _personApiGateway;
+        private readonly IAccountApiGateway _accountApiGateway;
+        private readonly IESEntityFactory _esEntityFactory;
+
+        public PersonBalanceUpdatedUseCase(IEsGateway esGateway, ITenureApiGateway tenureApiGateway, IPersonApiGateway personApiGateway,
+            IAccountApiGateway accountApiGateway, IESEntityFactory esPersonFactory)
+        {
+            _esGateway = esGateway;
+            _tenureApiGateway = tenureApiGateway;
+            _personApiGateway = personApiGateway;
+            _accountApiGateway = accountApiGateway;
+            _esEntityFactory = esPersonFactory;
+        }
+
+        public async Task ProcessMessageAsync(EntityEventSns message)
+        {
+            if (message is null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            //1. Get master account from Account service API
+            var account = await _accountApiGateway.GetAccountByIdAsync(message.EntityId).ConfigureAwait(false);
+            if (account is null)
+            {
+                throw new EntityNotFoundException<Account>(message.EntityId);
+            }
+
+            // Hanna Holasava
+            // Here TargetId is TenureId
+            // 2. Get Tenure from Tenure service API
+            var tenure = await _tenureApiGateway.GetTenureByIdAsync(account.TargetId).ConfigureAwait(false);
+            if (tenure is null)
+            {
+                throw new EntityNotFoundException<TenureInformation>(account.TargetId);
+            }
+
+            // Hanna Holasava
+            // Can we be sure that we will find only one record with IsResponsible = true? 
+            // If it is, we should change FirstOrDefault to SingleOrDefault.
```

Comment:
This is a good point - It may well be that we should use SingleOrDefault. @LBHMGeorgieva  ?

Reply:
With tenures, there could be multiple people where isResponsible=true as sometimes we have joint tenancy (where more than 1 person is the legally responsbile tenant) and in some other cases, more than 2 people can be responsible.

Reply:
@LBHMGeorgieva so given this how we could find person for some tenure?

Reply:
@AnnaGolosova depends, what are your requirements in your use case? Using isResponsible on it's own would not be sufficient if you want to identify a specific person as there could be more than one person where isResponsible=true